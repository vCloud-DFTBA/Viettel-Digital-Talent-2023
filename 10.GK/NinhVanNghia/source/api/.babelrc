{
  "presets": [
    [
      "@babel/preset-env",
      {
        "targets": {
          "node": "current"
        }
      }
    ]
  ],
  "plugins": [
    [
      "module-resolver",
      {
        "root": ["./src"]
      }
    ],
    "@babel/plugin-transform-runtime",
    "@babel/plugin-proposal-class-properties",
    "@babel/plugin-proposal-object-rest-spread"
  ],
  "minified": true,
  "comments": false
}
