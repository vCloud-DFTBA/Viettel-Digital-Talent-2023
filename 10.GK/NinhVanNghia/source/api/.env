HOSTNAME=localhost
PORT=8000

# Data Base
DB_HOST=localhost
DB_USERNAME=root
DB_PASSWORD=1234
DB_PORT=3306
DB_NAME=vdt
DB_DIALECT=mysql
DB_POOL_MIN=1
DB_POOL_MAX=5
DB_POOL_ACQUIRE=30000
DB_POOL_IDLE=10000
