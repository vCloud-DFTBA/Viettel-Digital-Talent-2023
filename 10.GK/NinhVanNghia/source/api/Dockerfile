FROM node:18-alpine as build

WORKDIR /app

COPY package.json .
COPY package-lock.json .

RUN npm install -f

COPY . .

RUN npm run build


### Final Stage
FROM node:18-alpine

RUN addgroup --gid 1001 -S vt_app && adduser --uid 1001 -G vt_app -S vt_app

WORKDIR /app

COPY package.json .
COPY package-lock.json .

RUN npm install --omit=dev -f

RUN chown -R vt_app:vt_app /app

USER vt_app

COPY --from=build /app/dist /app/dist

EXPOSE 8000

CMD ["npm", "start"]
