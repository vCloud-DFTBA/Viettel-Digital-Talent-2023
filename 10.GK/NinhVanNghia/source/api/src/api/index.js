import {Router} from 'express';

import { attendeeRouter } from './attendee/attendee.router';


export const apiRouter = Router();


apiRouter.use(attendeeRouter);
