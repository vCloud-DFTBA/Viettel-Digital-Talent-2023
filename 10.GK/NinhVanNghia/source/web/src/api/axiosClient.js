import axios from 'axios'
import {toast} from 'react-toastify'


const axiosClient = axios.create({
    baseURL: `${process.env.REACT_APP_API_URL || ''}/api/v1`,
});


axiosClient.interceptors.request.use(async (config) => {
    const {method, baseURL, url} = config
    console.log(`${new Date().toISOString()}  ${method.toUpperCase()}  ${baseURL + url}`);
    return config;
});

  

axiosClient.interceptors.response.use(
    ({data}) => {
        return data;
    },
    (error) => {
        const { data } = error.response || {};

        data?.message?.forEach(toast.error)
        !data && toast.error('Lỗi kết nối')

        throw error;
    }
);


export default axiosClient
