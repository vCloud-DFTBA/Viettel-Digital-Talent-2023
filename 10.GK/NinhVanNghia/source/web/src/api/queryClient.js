import { QueryClient } from '@tanstack/react-query';


const queryClient = new QueryClient({
    defaultOptions: {
        queries: {
            staleTime: Infinity,
            keepPreviousData: true,
            refetchOnReconnect: false,
            refetchOnWindowFocus: false,
            retry(failureCount, error) {
                return error.response
                    && !([404, 408, 409, 429, 502, 503, 504].includes(error.response.status))
                    && (failureCount < 3)
            },
        },
    },
})


export default queryClient;
