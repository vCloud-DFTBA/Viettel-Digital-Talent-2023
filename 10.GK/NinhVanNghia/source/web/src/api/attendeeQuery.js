import axiosClient from "./axiosClient";
import { useQuery, useMutation } from "@tanstack/react-query";


const attendeeKeys = {
    list() {
        return ['attendee']
    },
    get(attendeeId) {
        return [...this.list(), attendeeId]
    },
}


const attendeeQuery = {
    useList({query, params, body, options}={}) {
        return useQuery(
            attendeeKeys.list(),
            async () => {return await axiosClient.get(`/attendee`, {params: query, ...body})},
            options,
        )
    },

    useGet({query, params, body, options}={}) {
        return useQuery(
            attendeeKeys.get(),
            async () => {return await axiosClient.get(`/attendee/${params}`, {params: query, ...body})},
            options,
        )
    },
}


export {
    attendeeKeys,
    attendeeQuery,
}
