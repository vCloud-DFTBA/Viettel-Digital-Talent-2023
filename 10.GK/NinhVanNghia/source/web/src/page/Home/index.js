import AttendeeRow from "./components/AttendeeRow";
import CreateButton from "./components/CreateButton";

import {attendeeQuery} from "api/attendeeQuery";


function Home() {
    const {data, status} = attendeeQuery.useList({query: {page:0, limit: 100}})
    return (
        <div className="container">
            <div className="d-flex flex-row mb-1">
                <CreateButton className="ms-auto"/>
            </div>
            <div className="row border-bottom">
                <div className="col-3">Họ Và Tên</div>
                <div className="col-2">Năm sinh</div>
                <div className="col-5">Trườnng</div>
                <div className="col-2">thao tác</div>
            </div>
            {status === "success" && 
            data.attendees.map(item => <AttendeeRow {...item} key={item.attendeeId}/>)}
        </div>
    )
}


export default Home
