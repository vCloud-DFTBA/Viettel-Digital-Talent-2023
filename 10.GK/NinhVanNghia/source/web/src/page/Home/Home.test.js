import renderer from 'react-test-renderer';

import Home from '.';
import { attendeeQuery } from 'api/attendeeQuery';


test("It should render", async () => {
    jest.spyOn(attendeeQuery, 'useList').mockResolvedValue(Promise.resolve({}));
    expect(renderer.create(<Home/>).toJSON()).toMatchSnapshot()
})
