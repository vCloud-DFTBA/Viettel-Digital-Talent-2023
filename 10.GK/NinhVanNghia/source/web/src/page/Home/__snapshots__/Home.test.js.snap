// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It should render 1`] = `
<div
  className="container"
>
  <div
    className="d-flex flex-row mb-1"
  >
    <button
      className="ms-auto font-weight-bold btn btn-primary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Thêm người tham dự
    </button>
  </div>
  <div
    className="row border-bottom"
  >
    <div
      className="col-3"
    >
      Họ Và Tên
    </div>
    <div
      className="col-2"
    >
      Năm sinh
    </div>
    <div
      className="col-5"
    >
      Trườnng
    </div>
    <div
      className="col-2"
    >
      thao tác
    </div>
  </div>
</div>
`;
